### PR TITLE
fix #269467: added stringutils class

### PR DESCRIPTION
**Pull request by finleylau**
Added the stringutils class and added three tests (normal string that shouldn't be modified, one to replace all the ligatures, and one to replace the diacritics)

Continued from #3487

### CODE REVIEW

**Comment by lasconic** [comment]
Very cool ! Thanks!
Could you add some tests for CJK, russian, hebrew or arabic ?

File: mscore/stringutils.h [hunk 1]
```diff
@@ -0,0 +1,37 @@
+//=============================================================================
+//  MuseScore
+//  Linux Music Score Editor
+//  Copyright (C) 2002-2009 Werner Schweer and others
```
Comment (Jojo-Schmitz) [inline mscore/stringutils.h]: 2018 meanwhile ;-)

File: mscore/stringutils.cpp [hunk 2]
```diff
@@ -0,0 +1,90 @@
+//=============================================================================
+//  MuseScore
+//  Linux Music Score Editor
+//  Copyright (C) 2002-2009 Werner Schweer and others
```
Comment (Jojo-Schmitz) [inline mscore/stringutils.cpp]: 2018 meanwhile ;-)

**Comment by finleylau** [comment]
updated with tests for Greek, Chinese, Hebrew, and Arabic

File: mscore/stringutils.cpp [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+      // Big Letter OE (&#338) and small letter oe (&#339), O with above dots, o with above dots
+      result = result.replace(QRegularExpression("[\\x{0152}\\x{00D6}]"), QString("OE"));
+      result = result.replace(QRegularExpression("[\\x{0153}\\x{00F6}]"), QString("oe"));
+
+      // Big Letter OO (&#42830) and small letter oo (&#42831)
+      result = result.replace(QRegularExpression("[\\x{A74E}]"), QString("OO"));
+      result = result.replace(QRegularExpression("[\\x{A74F}]"), QString("oo"));
+
+      // ue (&#7531), U with above dots, u with above dots
+      result = result.replace(QRegularExpression("[\\x{00DC}]"), QString("UE"));
```
Comment (Jojo-Schmitz) [inline mscore/stringutils.cpp]: 'Ü' becomes `Ue`, not 'UE'

File: mscore/stringutils.cpp [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+
+      // Eszett (&#223)
+      result = result.replace(QRegularExpression("[\\x{00DF}]"), QString("ss"));
+
+      // O with stroke (&#216) and o with stroke (&#248)
+      result = result.replace(QRegularExpression("[\\x{00D8}]"), QChar('O'));
+      result = result.replace(QRegularExpression("[\\x{00F8}]"), QChar('o'));
+
+      // Big Letter OE (&#338) and small letter oe (&#339), O with above dots, o with above dots
+      result = result.replace(QRegularExpression("[\\x{0152}\\x{00D6}]"), QString("OE"));
```
Comment (Jojo-Schmitz) [inline mscore/stringutils.cpp]: 'Ö' becomes `Oe`, not 'OE'

File: mscore/stringutils.cpp [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+QString stringutils::removeLigatures(const QString& pre)
+      {
+      QString result = pre;
+
+      // Latin Big Letter AA (&#42802) and Latin Small Letter aa (&#42803)
+      result = result.replace(QRegularExpression("[\\x{A732}]"), QString("AA"));
+      result = result.replace(QRegularExpression("[\\x{A733}]"), QString("aa"));
+
+      // Latin Big Letter AE (&#198) and Latin Small Letter ae (&#230), A with above dots, a with above dots
+      result = result.replace(QRegularExpression("[\\x{00C6}\\x{00C4}]"), QString("AE"));
```
Comment (Jojo-Schmitz) [inline mscore/stringutils.cpp]: 'Ä' becomes `Ae`, not 'AE'

File: mscore/stringutils.cpp [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+      // Latin Big Letter AO (&#42804) and Latin Small Letter ao (&#42805)
+      result = result.replace(QRegularExpression("[\\x{A734}]"), QString("AO"));
+      result = result.replace(QRegularExpression("[\\x{A735}]"), QString("ao"));
+
+      // Latin Big Letter AU (&#42806) and Latin Small Letter au (&#42807)
+      result = result.replace(QRegularExpression("[\\x{A736}]"), QString("AU"));
+      result = result.replace(QRegularExpression("[\\x{A737}]"), QString("au"));
+
+      // IJ (&#306) and ij (&#307)
+      result = result.replace(QRegularExpression("[\\x{0132}]"), QString("IJ"));
```
Comment (Jojo-Schmitz) [inline mscore/stringutils.cpp]: This is indeed right, both chars uppercase

File: mscore/stringutils.cpp [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+      result = result.replace(QRegularExpression("[\\x{0133}]"), QString("ij"));
+
+      // Eszett (&#223)
+      result = result.replace(QRegularExpression("[\\x{00DF}]"), QString("ss"));
```
Comment (Jojo-Schmitz) [inline mscore/stringutils.cpp]: As of latest there is also an upper case `ẞ` , &#7838; (compare `ß`), to be used in upper case words, so there it should translate to `SS`

File: mscore/stringutils.cpp (hunk 3 above)
Comment (Jojo-Schmitz) [reply mscore/stringutils.cpp]: Not sure about the capitalization of  any of the other ligatures

**Comment by finleylau** [comment]
@Jojo-Schmitz I looked up capitalization of ligatures ([https://en.wikipedia.org/wiki/Capitalization#Digraphs_and_ligatures]) and according to this in general ligatures are usually capitalized and followed by a lowercase, so  I updated all the capitalized ligatures except for IJ and SS

**Pull request merged**